src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/idam/IdamTestService.java

```diff
@@ -58,37 +69,66 @@ public IdamTestService(
 
     public User createSolicitor() {
         String email = testData.nextUserEmail();
-        createUser(createSolicitorRequest(email, aatConfiguration.getSmokeTestSolicitor().getPassword()));
-        return userService.authenticateUser(email, aatConfiguration.getSmokeTestSolicitor().getPassword());
+        return Failsafe.with(retryPolicy)
+            .get(() -> {
+                try {
+                    idamTestApi.createUser(
+                        createSolicitorRequest(email, aatConfiguration.getSmokeTestCitizen().getPassword())
+                    );
+                } catch (ForbiddenException ex) {
+                    ex.printStackTrace();
+                }
+                return userService.authenticateUser(email, aatConfiguration.getSmokeTestSolicitor().getPassword());
+            });
```

Comment:
Can you extract these suppliers to some readable methods?
`return Failsafe.with(retryPolicy).get(() -> authenticateSolicitor(email));`
for example?

Reply:
will do